modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -1348,7 +1364,22 @@ private MediaPackage mergeScheduledMediaPackage(MediaPackage mp) throws IngestEx
     }
   }
 
-  private void mergeMediaPackageElements(MediaPackage mp, MediaPackage scheduledMp) {
+  private void mergeMediaPackageElements(final MediaPackage mp, final MediaPackage scheduledMp) {
+    // drop catalogs sent by the capture agent in favor of Opencast's own metadata
+    if (skipCatalogs) {
+      for (MediaPackageElement element : mp.getCatalogs()) {
+        mp.remove(element);
+      }
+    }
+
+    // drop attachments the capture agent sent us in favor of Opencast's attachments
+    // e.g. prevent capture agents from modifying security rules of schedules events
+    if (skipAttachments) {
+      for (MediaPackageElement element : mp.getAttachments()) {
+        mp.remove(element);
+      }
+    }
```

Comment:
@lkiesow Does it make sense to omit the opportunity for the ingest service merge new attachments and catalogs into the existing mediapackage while avoiding overwriting existing the catalogs and attachments (defined by flavor)?

Reply:
I'm not going to do that. I have/see no use case for that. Quite the contrary, I think it might even cause confusion about what will ultimately end up in the media package and it will add further code making the module even more complex. That said, I'm not against you adding that functionality if you have a specific use case for it. We just don't.